### DISCUSSION TITLE
Issues for beginners

### DISCUSSION THREAD

**Discussion by fabiante** [pdfcpu/pdfcpu#721] [Ideas]
Hi, I'd like to give contributing to pdfcpu a try. For that I want to propose: Can we try to label issues which should be simple in one sense or the other as "beginner friendly" or similar?

I have been coding Go for a couple of years, although more "simple" api stuff and generally some algorithms. I have some experience using various PDF processors and have also started to look into the underlying file format of PDF. I think anyone who has done that may have been at least _suprised_. 😉 

Since PDF is generally not a simple topic I assume a majority of issues are beginner friendly only in the sense that implementing / solving them is less dependant on lot's of mechanism, not beginner friendly as in "you learned go last week? give it a try".

@hhrutter what do you think?

**Comment by hhrutter** [comment]
Hi there!
I appreciate the effort and motivation - I really do but this is kind of a difficult topic.
pdfcpu is still Alpha and will be until we have covered the basics as laid out in #235.
Other than that I am having a hard time labelling any of the ongoing work as beginner friendly.
Addressing issues involves going back and forth between the various specifications involved and 
end up in a judgement call or many times in heavy refactoring.
Once we have reached Beta and the codebase is more stable - different story.
Let me think about this, there are some corners that may be worth a try like extending the annotations API.
Let's talk after the next release.
Thank you!

**Comment by hhrutter** [comment]
Closing this as answered.

I still think the original concern is valid: pdfcpu is not an easy codebase for “good first issue” style contributions because many apparently small changes involve PDF spec details, validation behavior, backwards compatibility, or larger refactoring.

For now I don’t want to introduce a beginner friendly label because it could create the wrong expectation. When I see an issue that is suitable for outside contribution, I will use help wanted instead and try to add enough context for someone to get started.

Thanks again for raising this.
